File: backend_golang/router/routes/signup.go
package routes

import (
	"backend_golang/constants"
	"backend_golang/router/sqldb"
	"backend_golang/utils"
	"encoding/json"
	"fmt"
	"net/http"

	_ "github.com/go-sql-driver/mysql"
)

//SignupHandler POST API
func SignupHandler(res http.ResponseWriter, req *http.Request) {

	utils.SetHeaders(&res)
	response := make(map[string]string)

	if sqldb.DB == nil {
		response["ModalTitle"] = "Service Unavailable..."
		response["ModalBody"] = "Signup Service is unavailable right now... Please try again later"
		res.WriteHeader(http.StatusServiceUnavailable)
		panic("503 Error")
	}

	defer func() {
		if r := recover(); r != nil {
			_ = json.NewEncoder(res).Encode(response)
		}
	}()

	_ = req.ParseMultipartForm(0)

	firstname := req.Form.Get("firstname")
	lastname := req.Form.Get("lastname")
	email := req.Form.Get("email")
	mobile := req.Form.Get("mobile")
	password := req.Form.Get("password")

	stmt := fmt.Sprintf("select * from %s.user where email = '%s';", constants.MYSQL_DATABASE, email)
	rows, _ := sqldb.DB.Query(stmt)
	defer rows.Close()

	userExists := rows.Next()
	if userExists {
		response["ModalTitle"] = "Email already registered..."
		response["ModalBody"] = "A User has already registered with the specified Email..Please try again with another Email..."
		res.WriteHeader(http.StatusForbidden)
		panic("403 Error")

	} else {
		stmt = fmt.Sprintf("insert into %s.user values(NULL,'%s','%s','%s','%s','%s',NULL);", constants.MYSQL_DATABASE, firstname, lastname, email, mobile, password)
		_, err := sqldb.DB.Query(stmt)
		if err != nil {
			response["ModalTitle"] = "Registration Failed..."
			response["ModalBody"] = "Error occured during registration..Please try again later..."
			res.WriteHeader(http.StatusForbidden)
			fmt.Println(err.Error())
			panic("403 Error")
		} else {
			response["ModalTitle"] = "Registration Successful..."
			response["ModalBody"] = "Registration has been completed successfully..Please Login..."
			res.WriteHeader(http.StatusCreated)
		}
	}
	_ = json.NewEncoder(res).Encode(response)
}

File: api/signup.php
<?php
    require("./config/php_cors.php");
    require("./config/connection.php");

    $response = array();

    try {

        $conn = (new DatabaseConnection())->getConnection();
        
        if(!$conn){
            http_response_code(503);
            $response["ModalTitle"]="Service Unavailable...";
            $response["ModalBody"]="Signup Service is unavailable right now... Please try again later";
            throw new Exception("503 Error"); 
        }

        if($_SERVER["REQUEST_METHOD"]!="POST"){
            http_response_code(503);
            $response["ModalTitle"]="Method Not Allowed...";
            $response["ModalBody"]="Specified request type is not allowed";
            throw new Exception("503 Error"); 
        }

        $firstname = $_POST['firstname'];
        $lastname = $_POST['lastname'];
        $email = $_POST['email'];
        $mobile = $_POST['mobile'];
        $password = $_POST['password'];


        $sql = "select * from user where email='$email';";
        $res = $conn->query($sql);

        if($res->num_rows>0){
            http_response_code(403);
            $response["ModalTitle"]="Email already registered...";
            $response["ModalBody"]="A User has already registered with the specified Email..Please try again with another Email...";
            throw new Exception("403 Error");
        }

        $sql = "insert into user values(NULL,'$firstname','$lastname','$email','$mobile','$password',NULL);";
        if($conn->query($sql)){
            $response["ModalTitle"]="Registration Successful...";
            $response["ModalBody"]="Registration has been completed successfully..Please Login...";
            http_response_code(201);
        }else{
            $response["ModalTitle"]="Registration Failed...";
            $response["ModalBody"]="Error occured during registration..Please try again later...";    
            http_response_code(403);
            throw new Exception("403 Error");    
        }

        echo json_encode($response);
    
    }catch(Exception $exception){
        echo json_encode($response);
    }
?>